pennylane/ops/op_math/exp.py

```diff
@@ -227,20 +228,24 @@ def decomposition(self):
         return [qml.PauliRot(new_coeff, string_base, wires=self.wires)]
 
     def matrix(self, wire_order=None):
-
-        coeff_interface = math.get_interface(self.coeff)
-        if coeff_interface == "autograd" and math.requires_grad(self.coeff):
+        scalar = [self.scalar] if qml.math.size(self.scalar) == 1 else self.scalar
```

Comment:
Wait, now you're making me doubt... I believe my code is wrong:

```pycon
>>> scalar = [1]
>>> [scalar] if qml.math.size(scalar) == 1 else scalar
[[1]]
```

I should use a `qml.math.ndim` check here.

Reply:
Changed to `qml.math.ndim` 👍

Reply:
I am utterly mystified because I am about 99.9% sure I tested my hunch correctly before making this comment, but I'm definitely seeing [[1]] now... glad I said something!